File: problem-solving/algorithms/grading-students/grading-students.js
function gradingStudents(grades) {
    const roundedGrades = [];

    for (let grade of grades) {
        if (grade < 38) {
            roundedGrades.push(grade);
        } else {
            const nextMultipleOfFive = Math.ceil(grade / 5) * 5;

            if (nextMultipleOfFive - grade < 3) {
                roundedGrades.push(nextMultipleOfFive);
            } else {
                roundedGrades.push(grade);
            }
        }
    }

    return roundedGrades;
}

File: problem-solving/algorithms/grading-students/grading_students.py
def gradingStudents(grades):
    rounded_grades = []

    for grade in grades:
        if grade < 38:
            rounded_grades.append(grade)
        else:
            next_multiple_of_five = (grade // 5 + 1) * 5

            if next_multiple_of_five - grade < 3:
                rounded_grades.append(next_multiple_of_five)
            else:
                rounded_grades.append(grade)

    return rounded_grades